tools/generate_hotkey.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+from collections import defaultdict
+
+from zulipterminal.config.keys import HELP_CATEGORIES, KEY_BINDINGS
+
+
+# category = {}
+# {category.setdefault(item['key_category'],[]).append((item['help_text'], item['keys'])) for (key, item) in KEY_BINDINGS.items()}
+
+# OR
+category = defaultdict(list)
+{category[item['key_category']].append((item['help_text'], item['keys'])) for (key, item) in KEY_BINDINGS.items()}
```

Comment:
Comprehensions generally build an object itself and don't use side-effects (walrus operator aside), so I'd expect this dict to be assigned to `category` or some other name.

You may need a nested comprehension, or another data-structure may be better.

Reply:
> You may need a nested comprehension, or another data-structure may be better.

Sorry for the delay. 
I have used a nested comprehension this time. Also, I would like to know which alternate data structure you were referring to here.

Reply:
What you have now looks fine :+1: